virtual_rainforest/models/abiotic_simple/simple_regression.py

```diff
@@ -0,0 +1,571 @@
+    all atmospheric levels.
+
+    Soil moisture and surface runoff are calculated with a simple bucket model based
+    on :cite:t:`davis_simple_2017`: if
+    precipitation exceeds soil moisture capacity (see
+    :data:`~virtual_rainforest.models.abiotic_simple.simple_regression.MicroclimateParameters`)
+    , the excess water is added to runoff and soil moisture is set to soil moisture
+    capacity value; if the soil is not saturated, precipitation is added to the current
+    soil moisture level and runoff is set to zero.
+
+    The `layer_roles` list is composed of the following layers (index 0 above canopy):
+
+    * above canopy (canopy height + reference measurement height, typically 2m)
+    * canopy layers (maximum of ten layers, minimum one layers)
+    * subcanopy (1.5 m)
+    * surface layer
+    * soil layers (currently one near surface layer and one layer at 1 m below ground)
+
+    The function expects a data object with the following variables:
+
+    * air_temperature_ref
+    * relative_humidity_ref
+    * atmospheric_pressure_ref
+    * atmospheric_co2_ref
+    * leaf_area_index
+    * layer_heights
+    * precipitation
+    * soil_moisture
+
+    Args:
+        data: Data object
+        layer_roles: list of layer roles (from top to bottom: above, canopy, subcanopy,
+            surface, soil)
+        time_index: time index, integer
+        MicroclimateGradients: gradients for linear regression
+            :cite:p:`hardwick_relationship_2015`
+        Bounds: upper and lower allowed values for vertical profiles, used to constrain
+            log interpolation. Note that currently no conservation of water and energy!
+        water_interception_factor: Factor that determines how much rainfall is
+            intercepted by stem and canopy
+        soil_moisture_capacity: soil moisture capacity for water
+
+    Returns:
+        Dict of DataArrays for air temperature [C], relative humidity [-], vapour
+        pressure deficit [kPa], soil temperature [C], atmospheric pressure [kPa],
+        atmospheric :math:`\ce{CO2}` [ppm], soil moisture [-], and surface runoff [mm]
+    """
+
+    # TODO correct gap between 1.5 m and 2m reference height for LAI = 0
+    # TODO make sure variables are representing correct time interval, e.g. mm per day
+    output = {}
+
+    leaf_area_index_sum = data["leaf_area_index"].sum(dim="layers")
+
+    # Mean air temperature profile, [C]
+    output["air_temperature"] = log_interpolation(
+        data=data,
+        reference_data=data["air_temperature_ref"].isel(time=time_index),
+        leaf_area_index_sum=leaf_area_index_sum,
+        layer_roles=layer_roles,
+        layer_heights=data["layer_heights"],
+        upper_bound=Bounds["air_temperature_max"],
+        lower_bound=Bounds["air_temperature_min"],
+        gradient=MicroclimateGradients["air_temperature_gradient"],
+    ).rename("air_temperature")
```

Comment:
Several of these blocks are repeated. I'd be tempted to restructure/rename elements so that you can use a single variable name in a loop. Something like:

```python
for vars in ['air_temperature', ...]:

    output[var] = log_interpolation(
        data=data,
        reference_data=data[var + "_ref"].isel(time=time_index),
        leaf_area_index_sum=leaf_area_index_sum,
        layer_roles=layer_roles,
        layer_heights=data["layer_heights"],
        upper_bound=Bounds[var].max,
        lower_bound=Bounds[var].min,  # Or even just bounds = Bounds[var]
        gradient=MicroclimateGradients[var],
    ).rename(var)

Reply:
That function is also unpacking nearly everything it needs from `data` into arguments, but still passing in `data`. You could just pass in data and the `var` name and extract internally, or pass in all the required arguments ready extracted - I think it is just `cell_id` you're getting from `data`?

Reply:
I combined `air_temperature`, `relative_humidity` and `vapour_pressure_deficit` in a loop. To make it work, I moved the calculation of the reference vapour pressure deficit to the setup (it can be done for all time steps in one go)